resources/snakefiles/binning.smk

```diff
@@ -181,18 +191,103 @@ rule make_concoct_coverage_table:
 
     """
     input:
-        bed="output/binning/concoct/{mapper}/contigs_10K/{contig_sample}.bed",
-        bams = lambda wildcards: get_bam_list(wildcards.contig_sample, config['mappers'], contig_pairings)
+        bed = "output/binning/concoct/{mapper}/contigs_10K/{contig_sample}.bed",
+        bam = lambda wildcards: get_bam_list(wildcards.contig_sample, config['mappers'], contig_pairings),
+        index = lambda wildcards: get_index_list(wildcards.contig_sample, config['mappers'], contig_pairings)
     output:
-        coverage_table="output/binning/concoct/{mapper}/coverage_tables/{contig_sample}_coverage_table.txt"
+        coverage_table = "output/binning/concoct/{mapper}/coverage_tables/{contig_sample}_coverage_table.txt"
     conda:
-        "../env/binning_concoct_osx.yaml"
+        "../env/concoct_linux.yaml"
     benchmark:
         "output/benchmarks/binning/concoct/{mapper}/make_concoct_coverage_table/{contig_sample}_benchmark.txt"
     log:
         "output/logs/binning/concoct/{mapper}/make_concoct_coverage_table/{contig_sample}.log"
     shell:
         """
-          python resources/scripts/concoct_coverage_table.py {input.bed} \
-          {input.bams} > {output.coverage_table} 2> {log}
+          concoct_coverage_table.py {input.bed} \
+          {input.bam} > {output.coverage_table} 2> {log}
+          touch {input.bam}
```

Comment:
I would delete these touch statements -- they're going to update the timestamp of the *input* files, which will cause all sorts of havoc with Snakemake's job constructor (every time this rule runs, the input files will have newer timestamps than the output files)

Reply:
If you don't have these touch statements, the index gets created by the `sort_index_bam` rule and then immediately gets deleted because it isn't explicitly used in any rule. 

`concoct_coverage_table.py` doesn't use indexes as an input, it just expects it to be in the  directory of the sorted bams.

Reply:
ok, done.